aiohasupervisor/models/discovery.py

```diff
@@ -0,0 +1,40 @@
+"""Models for discovery component."""
+
+from dataclasses import dataclass
+from typing import Any
+from uuid import UUID
+
+from .base import Request, ResponseData
+
+
+@dataclass(frozen=True, slots=True)
+class DiscoveryConfig(Request):
+    """DiscoveryConfig model."""
+
+    service: str
+    config: dict[str, Any]
+
+
+@dataclass(frozen=True, slots=True)
+class Discovery(ResponseData):
+    """Discovery model."""
+
+    addon: str
+    service: str
+    uuid: UUID
+    config: dict[str, Any]
+
+
+@dataclass(frozen=True, slots=True)
+class DiscoveryList(ResponseData):
+    """DiscoveryList model."""
+
+    discovery: list[Discovery]
+    services: dict[str, list[str]]
```

Comment:
Alternatively we could just provide one attribute that is a dict with service keys that maps to list of `Discovery` instances. It feels like we're duplicating data somewhat now.

Reply:
So these are actually different. But I'm honestly not sure who (if anyone) uses `services`.

Essentially `services` is showing you all the services you could be using, even if you aren't right now. And for each telling you which addons are available that provide that service.

Supervisor creates that dictionary by going through all the addons in the store (regardless of whether they are installed). For each unique service it finds listed in `discovery` of the config of these addons it adds a key. The value of that key is a list of addon slugs which say they provide that service in their `discovery` config.

So I'm not sure these two can really be merged. One is a list of active discovery items provided by addons running right now. The other is basically a directory of all the stuff you could be using and where to find it. We could try dropping `services` until it is requested though?

Reply:
Yea looking at core, we currently retrieve discovery messages from supervisor here:
https://github.com/home-assistant/core/blob/8d6f9cec7b560e37d20849462feee35b5695376e/homeassistant/components/hassio/handler.py#L387-L392

This method is used in two places within core. Both of them pull the list out of `discovery` and ignore `services:
https://github.com/home-assistant/core/blob/8d6f9cec7b560e37d20849462feee35b5695376e/homeassistant/components/hassio/handler.py#L111-L116
https://github.com/home-assistant/core/blob/8d6f9cec7b560e37d20849462feee35b5695376e/homeassistant/components/hassio/discovery.py#L44-L57

So I don't think anyone would mind if `services` got cut here. Perhaps we could deprecate it supervisor side as well.

Reply:
Ok. I guess there could be custom integrations using it. But it seems not very interesting from a Core perspective. Dropping it from the client here seems ok since we can always add it later (unless it's deprecated).

Reply:
Yea I think I'll just remove it from here and make this the "list discovery messages" method. If someone does need this information I'll make a new `info` method that returns the whole object including `services` and just have it call the same endpoint behind the scenes. But I think listing discovery is really all we need here.